src/ape/api/accounts.py

```diff
@@ -273,6 +278,10 @@ def check_signature(
         Returns:
             bool: ``True`` if the data was signed by this account. ``False`` otherwise.
         """
+        if isinstance(data, str):
+            data = encode_defunct(text=data)
+        if isinstance(data, EIP712Message):
```

Comment:
Should these all be `elif` ?

Reply:
the check for EIP712Message can be an elif (as well as the newly added int check), not the subsequent check for SignableMessage/EIP712SignableMessage